### PR TITLE
community[minor]:feat(embedding integration):  modifying the Alibaba Tongyi chat model enableSearch parameter is invalid; add Alibaba Tongyi embedding

### PR DESCRIPTION
**Pull request by JoeABCDEF**
Fix the issue that the chatModel searchEnable parameter is invalid when only using community alibaba_tongyi;
Embedding with alibaba_tongyi integration

### CODE REVIEW

**Comment by JoeABCDEF** [comment]
Could you please reply, Does the code not meet the specification? Why can't it be merged?

File: libs/langchain-community/src/embeddings/alibaba_embedding.ts
Comment (jacoblee93) [inline libs/langchain-community/src/embeddings/alibaba_embedding.ts:1]: Can you rename the file to `embeddings/alibaba_tongyi.ts` to match the chat model name?

https://github.com/langchain-ai/langchainjs/blob/main/libs/langchain-community/src/chat_models/alibaba_tongyi.ts

File: libs/langchain-community/src/embeddings/alibaba_embedding.ts [hunk 1]
```diff
@@ -0,0 +1,215 @@
+import { getEnvironmentVariable } from "@langchain/core/utils/env"
+import { Embeddings, type EmbeddingsParams } from "@langchain/core/embeddings"
+import { chunkArray } from "@langchain/core/utils/chunk_array"
+
+export interface AlibabaAIEmbeddingsParams extends EmbeddingsParams {
```
Comment (jacoblee93) [inline libs/langchain-community/src/embeddings/alibaba_embedding.ts:5]: Rename to `AlibabaTongyiEmbeddingsParams` to match naming precedent

File: libs/langchain-community/src/embeddings/alibaba_embedding.ts [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+    request_id: string
+}
+
+interface EmbeddingErrorResponse {
+    code: string
+    message: string
+    request_id: string
+}
+
+export class AlibabaAIEmbeddings extends Embeddings implements AlibabaAIEmbeddingsParams {
```
Comment (jacoblee93) [inline libs/langchain-community/src/embeddings/alibaba_embedding.ts:71]: See above, rename to `AlibabaTongyiEmbeddingsParams`

File: libs/langchain-community/src/embeddings/alibaba_embedding.ts [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+    batchSize = 24
+
+    // TODO: Update to `false` on next minor release (see: https://github.com/langchain-ai/langchainjs/pull/3612)
+    stripNewLines = true
+
+    /**
+     * The number of dimensions the resulting output embeddings should have.
+     * Only supported in `text-embedding-3` and later models.
+     */
+    // dimensions?: number
```
Comment (jacoblee93) [inline libs/langchain-community/src/embeddings/alibaba_embedding.ts:82]: Remove unused comments

File: libs/langchain-community/src/embeddings/alibaba_embedding.ts [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+
+        this.modelName = fieldsWithDefaults?.modelName ?? this.modelName
+        this.batchSize = fieldsWithDefaults?.batchSize ?? this.batchSize
+        this.stripNewLines = fieldsWithDefaults?.stripNewLines ?? this.stripNewLines
+
+        this.parameters = {
+            text_type: fieldsWithDefaults?.parameters?.text_type ?? "document"
+        }
+        // this.timeout = fieldsWithDefaults?.timeout
+        // this.dimensions = fieldsWithDefaults?.dimensions
```
Comment (jacoblee93) [inline libs/langchain-community/src/embeddings/alibaba_embedding.ts:114]: Remove commented out code

File: libs/langchain-community/src/embeddings/alibaba_embedding.ts [hunk 5]
```diff
@@ -0,0 +1,215 @@
...
+    }
+
+    /**
+     * Method to generate embeddings for an array of documents. Splits the
+     * documents into batches and makes requests to the OpenAI API to generate
```
Comment (jacoblee93) [inline libs/langchain-community/src/embeddings/alibaba_embedding.ts:119]: Update `OpenAI` in docstring

**Comment by jacoblee93** [comment]
Hey @JoeABCDEF, sorry for the delay, I have a lot to cover!

I don't think I can test this myself but the URL looks right and overall looks good, minus comments. Can you run `yarn format` and `yarn lint` and add a documentation page?

**Comment by JoeABCDEF** [comment]
> Hey @JoeABCDEF, sorry for the delay, I have a lot to cover!
> 
> I don't think I can test this myself but the URL looks right and overall looks good, minus comments. Can you run `yarn format` and `yarn lint` and add a documentation page?

I'm very sorry, a lot of things are not careful, but this function is normal that I have used it myself, and I will carefully change and resubmit the problem that occurs, I am very sorry

File: libs/langchain-community/src/embeddings/alibaba_embedding.ts
Comment (JoeABCDEF) [reply libs/langchain-community/src/embeddings/alibaba_embedding.ts:1]: I changed it

File: libs/langchain-community/src/embeddings/alibaba_embedding.ts (hunk 1 above)
Comment (JoeABCDEF) [reply libs/langchain-community/src/embeddings/alibaba_embedding.ts]: I changed it

File: libs/langchain-community/src/embeddings/alibaba_embedding.ts (hunk 2 above)
Comment (JoeABCDEF) [reply libs/langchain-community/src/embeddings/alibaba_embedding.ts]: I changed it

File: libs/langchain-community/src/embeddings/alibaba_embedding.ts (hunk 3 above)
Comment (JoeABCDEF) [reply libs/langchain-community/src/embeddings/alibaba_embedding.ts]: I changed it

File: libs/langchain-community/src/embeddings/alibaba_embedding.ts (hunk 4 above)
Comment (JoeABCDEF) [reply libs/langchain-community/src/embeddings/alibaba_embedding.ts]: I changed it

File: libs/langchain-community/src/embeddings/alibaba_embedding.ts (hunk 5 above)
Comment (JoeABCDEF) [reply libs/langchain-community/src/embeddings/alibaba_embedding.ts]: I changed it

**Comment by jacoblee93** [comment]
Added docs + entrypoint from `@langchain/community/embeddings/alibaba_tongyi` - thank you!

**Comment by JoeABCDEF** [comment]
> Added docs + entrypoint from `@langchain/community/embeddings/alibaba_tongyi` - thank you!

Okay, I will

**Comment by jacoblee93** [comment]
Oh, I added it! It's live:

https://js.langchain.com/docs/integrations/text_embedding/alibaba_tongyi

**Comment by JoeABCDEF** [comment]
> Oh, I added it! It's live:
> 
> https://js.langchain.com/docs/integrations/text_embedding/alibaba_tongyi
> 
> It is finished :)

thank you so much

**Comment by Jack-Z-Coding** [comment]
by the way ,how can you get the AlibabaTongyiEmbeddings key，when i visit "https://dashvector.console.aliyun.com/cn-hangzhou/overview" and get get key . its invalid
image

**Pull request merged**